File: modules/react-native-status/android/src/main/java/im/status/NimStatus.java
package im.status;





public class NimStatus {

  public NimStatus() {

  }
  
  public native String hashMessage(String message);
  public native String initKeystore(String keydir);
  public native String openAccounts(String datadir);
  public native String multiAccountGenerateAndDeriveAddresses(String paramsJSON);
  public native String multiAccountStoreDerivedAccounts(String paramsJSON);
  public native String multiAccountImportMnemonic(String paramsJSON);
  public native String multiAccountImportPrivateKey(String paramsJSON);
  public native String multiAccountDeriveAddresses(String paramsJSON);
  public native String saveAccountAndLogin(String accountData, String password, String settingsJSON, String configJSON, String subaccountData);
  public native String callRPC(String inputJSON);
  public native String callPrivateRPC(String inputJSON);
  public native String addPeer(String peer);
  public native void setSignalEventCallback(Object obj);
  public native String sendTransaction(String jsonArgs, String password);
  public native String generateAlias(String pk);
  public native String identicon(String pk);
  public native String login(String accountData, String password);
  public native String logout();
  public native String verifyAccountPassword(String keyStoreDir, String address, String password);
  public native String validateMnemonic(String mnemonic);
  public native String recoverAccount(String password, String mnemonic);
  public native String startOnboarding(int n, int mnemonicPhraseLength);
  public native String saveAccountAndLoginWithKeycard(String accountData, String password, String settingsJSON, String configJSON, String subaccountData, String keyHex);
  public native String hashTransaction(String txArgsJSON);
  public native String extractGroupMembershipSignatures(String signaturePairsStr);
  public native void connectionChange(String typ, String expensive);
  public native String multiformatSerializePublicKey(String key, String outBase);
  public native String multiformatDeserializePublicKey(String key, String outBase);
  public native String validateNodeConfig(String configJSON);
  public native String loginWithKeycard(String accountData, String password, String keyHex);
  public native String recover(String rpcParams);
  public native String writeHeapProfile(String dataDir);
  public native String importOnboardingAccount(String id, String password);
  public native void removeOnboarding();
  public native String hashTypedData(String data);
  public native String resetChainData();
  public native String signMessage(String rpcParams);
  public native String signTypedData(String data, String address, String password);
  public native String stopCPUProfiling();
  public native String getNodesFromContract(String rpcEndpoint, String contractAddress);
  public native String exportNodeLogs();
  public native String chaosModeUpdate(int on);
  public native String signHash(String hexEncodedHash);
  public native String createAccount(String password);
  public native String sendTransactionWithSignature(String txtArgsJSON, String sigString);
  public native String startCPUProfile(String dataDir);
  public native void appStateChange(String state);
  public native String signGroupMembership(String content);
  public native String multiAccountStoreAccount(String paramsJSON);
  public native String multiAccountLoadAccount(String paramsJSON);
  public native String multiAccountGenerate(String paramsJSON);
  public native String multiAccountReset();
  public native String deleteMultiaccount(String keyUID, String path);
  public native String migrateKeyStoreDir(String accountData, String password, String oldKeystoreDir, String multiaccountKeystoreDir);
  public native void startWallet();
  public native void stopWallet();
}

File: nix/nim-status/nim_status.nim
import lib/shim as nim_shim


import go/shim as go_shim

let hashMessage {.exportc.} = go_shim.hashMessage
let generateAlias {.exportc.} = go_shim.generateAlias
let identicon {.exportc.} = go_shim.identicon
let initKeystore {.exportc.} = go_shim.initKeystore
let openAccounts {.exportc.} = go_shim.openAccounts
let multiAccountGenerateAndDeriveAddresses {.exportc.} = go_shim.multiAccountGenerateAndDeriveAddresses
let multiAccountStoreDerivedAccounts {.exportc.} = go_shim.multiAccountStoreDerivedAccounts
let multiAccountImportMnemonic {.exportc.} = go_shim.multiAccountImportMnemonic
let multiAccountImportPrivateKey {.exportc.} = go_shim.multiAccountImportPrivateKey
let multiAccountDeriveAddresses {.exportc.} = go_shim.multiAccountDeriveAddresses
let saveAccountAndLogin {.exportc.} = go_shim.saveAccountAndLogin
let deleteMultiAccount {.exportc.} = go_shim.deleteMultiAccount
let callRPC {.exportc.} = go_shim.callRPC
let callPrivateRPC {.exportc.} = go_shim.callPrivateRPC
let addPeer {.exportc.} = go_shim.addPeer
let sendTransaction {.exportc.} = go_shim.sendTransaction
let login {.exportc.} = go_shim.login
let logout {.exportc.} = go_shim.logout
let verifyAccountPassword {.exportc.} = go_shim.verifyAccountPassword
let validateMnemonic {.exportc.} = go_shim.validateMnemonic
let saveAccountAndLoginWithKeycard {.exportc.} = go_shim.saveAccountAndLoginWithKeycard
let hashTransaction {.exportc.} = go_shim.hashTransaction
let extractGroupMembershipSignatures {.exportc.} = go_shim.extractGroupMembershipSignatures
let connectionChange {.exportc.} = go_shim.connectionChange
let multiformatSerializePublicKey {.exportc.} = go_shim.multiformatSerializePublicKey
let multiformatDeserializePublicKey {.exportc.} = go_shim.multiformatDeserializePublicKey
let validateNodeConfig {.exportc.} = go_shim.validateNodeConfig
let loginWithKeycard {.exportc.} = go_shim.loginWithKeycard
let recover {.exportc.} = go_shim.recover
let writeHeapProfile {.exportc.} = go_shim.writeHeapProfile
let hashTypedData {.exportc.} = go_shim.hashTypedData
let resetChainData {.exportc.} = go_shim.resetChainData
let signMessage {.exportc.} = go_shim.signMessage
let signTypedData {.exportc.} = go_shim.signTypedData
let stopCPUProfiling {.exportc.} = go_shim.stopCPUProfiling
let getNodesFromContract {.exportc.} = go_shim.getNodesFromContract
let exportNodeLogs {.exportc.} = go_shim.exportNodeLogs
let chaosModeUpdate {.exportc.} = go_shim.chaosModeUpdate
let signHash {.exportc.} = go_shim.signHash
let sendTransactionWithSignature {.exportc.} = go_shim.sendTransactionWithSignature
let startCPUProfile {.exportc.} = go_shim.startCPUProfile
let appStateChange {.exportc.} = go_shim.appStateChange
let signGroupMembership {.exportc.} = go_shim.signGroupMembership
let multiAccountStoreAccount {.exportc.} = go_shim.multiAccountStoreAccount
let multiAccountLoadAccount {.exportc.} = go_shim.multiAccountLoadAccount
let multiAccountGenerate {.exportc.} = go_shim.multiAccountGenerate
let multiAccountReset {.exportc.} = go_shim.multiAccountReset
let migrateKeyStoreDir {.exportc.} = go_shim.migrateKeyStoreDir
let startWallet {.exportc.} = go_shim.startWallet
let stopWallet {.exportc.} = go_shim.stopWallet
let startLocalNotifications {.exportc.} = go_shim.startLocalNotifications
let stopLocalNotifications {.exportc.} = go_shim.stopLocalNotifications

type SignalCallback {.exportc: "SignalCallback".} = proc(eventMessage: cstring): void {.cdecl.}

proc setSignalEventCallback(callback: SignalCallback) {.exportc.} =
  go_shim.setSignalEventCallback(callback)

# proc setupForeignThreadGc {.exportc.} = 
#   echo ""
# proc tearDownForeignThreadGc {.exportc.} = 
#   echo ""
let setupForeignThreadGc {.exportc.} = (proc = discard)
let tearDownForeignThreadGc {.exportc.} = (proc = discard)

